packages/types/src/api/web/deployment.ts

```diff
@@ -0,0 +1,14 @@
+import { DeploymentDoc, DeploymentStatus } from "../../documents"
+
+export interface PublishAppResponse extends DeploymentDoc {}
+
+export type DeploymentProgressResponse =
+  | {
+      _id: string
+      appId: string
+      status?: DeploymentStatus
+      updatedAt: number
+    }
+  | undefined
```

Comment:
Should undefined be part of the usage, instead of the type?

Reply:
In this case - this API isn't used today, so I'll make a small change so the API can't return undefined, but I would say if the API can return nothing/undefined then its worth capturing in the types so the frontend knows.